cpp/src/strings/regex/regcomp.cpp

```diff
@@ -731,26 +731,39 @@ class regex_parser {
         CUDF_EXPECTS(repeat_start_index >= 0, "regex: invalid counted quantifier location");
 
         // range of affected item(s) to repeat
-        auto const begin = in.begin() + repeat_start_index;
-        auto const end   = in.begin() + index;
+        auto const begin = out.begin() + repeat_start_index;
+        auto const end   = out.end();
+
         // count range values
         auto const n = item.d.count.n;  // minimum count
         auto const m = item.d.count.m;  // maximum count
-
         assert(n >= 0 && "invalid repeat count value n");
         // zero-repeat edge-case: need to erase the previous items
-        if (n == 0) { out.erase(out.end() - (index - repeat_start_index), out.end()); }
-
-        // minimum repeats (n)
-        for (int j = 1; j < n; j++) {
-          out.insert(out.end(), begin, end);
+        if (n == 0 && m == 0) { out.erase(begin, end); }
+
+        std::vector<regex_parser::Item> repeat_copy(begin, end);
+        // special handling for quantified capture groups
+        if ((n > 1) && (*begin).type == LBRA) {
+          (*begin).type = LBRA_NC;  // change first one to non-capture
+          // add intermediate groups as non-capture
```

Comment:
Why do we make these non-capturing? Do we handle capturing in some other way that isn't local to this code so I'm not seeing it, or is there some reason that we're guaranteed that these will never be capturing groups (or is this just a case that we explicitly don't support)?

Reply:
The behavior is an attempt to honor the regex spec defined in the first paragraph here
https://www.regular-expressions.info/captureall.html
Specifically this phrase: _repeated capturing group will capture only the last iteration_
So all the intermediate groups are converted to non-capture leaving only the last iteration as capturing.
Example from #16730: https://regex101.com/r/QdDTVZ/1